### PR TITLE
[CN-717] Ingress support for Management Center

### PR DESCRIPTION
**Pull request by semihbkgr**
## Description

In addition to service for Mancenter external connectivity, k8s Ingress resource should be added to Mancener.

## User Impact

Ingress support for Management Center external connectivity.

### CODE REVIEW

**Review by mtyazici — APPROVED** [review]
LGTM :+1: , just left a few comments

File: test/integration/managementcenter_controller_test.go [hunk 1]
```diff
@@ -160,6 +174,110 @@ var _ = Describe("ManagementCenter controller", func() {
...
+			mc := &hazelcastv1alpha1.ManagementCenter{
+				ObjectMeta: GetRandomObjectMeta(),
+				Spec:       hazelcastv1alpha1.ManagementCenterSpec{},
+			}
+
+			Create(mc)
+			fetchedMc := EnsureStatus(mc)
+			test.CheckManagementCenterCR(fetchedMc, defaultSpecValues, ee)
+
+			expectedOwnerReference := metav1.OwnerReference{
```
[the remark this answers is not in the document]
Comment (semihbkgr) [reply test/integration/managementcenter_controller_test.go:190]: It is to be sure that ownerReference is not changed after some updates, but it is not needed much

File: test/integration/managementcenter_controller_test.go [hunk 2]
```diff
@@ -160,6 +174,110 @@ var _ = Describe("ManagementCenter controller", func() {
...
+			assertDoesNotExist(lookupKey(mc), ing)
+
+			externalConnectivityIngress := &hazelcastv1alpha1.ExternalConnectivityIngress{
+				IngressClassName: "nginx",
+				Annotations:      map[string]string{"app": "hazelcast-mc"},
+				Hostname:         "mancenter",
+			}
+			fetchedMc.Spec.ExternalConnectivity.Ingress = externalConnectivityIngress
+
+			expectedOwnerReference := metav1.OwnerReference{
```
[the remark this answers is not in the document]
Comment (semihbkgr) [reply test/integration/managementcenter_controller_test.go:235]: It is not same as above. In this case, ingress' ownerReference is checking.

File: internal/phonehome/phonehome.go [hunk 3]
```diff
@@ -148,6 +156,7 @@ func newPhoneHomeData(cl client.Client, m *Metrics) PhoneHomeData {
 	phd.fillReplicatedMapMetrics(cl)
 	phd.fillCronHotBackupMetrics(cl)
 	phd.fillTopicMetrics(cl)
+	phd.fillMcExternalConnectivityMetrics(cl)
```
Comment (mtyazici) [inline internal/phonehome/phonehome.go:159]: Can you call this inside `fillMCMetrics` like Hazelcast does for `exposeExternally`?

**Review by kutluhanmetin — APPROVED** [review]

**Pull request merged**